src/storage-controller/src/collection_mgmt.rs

```diff
@@ -198,6 +201,9 @@ where
                     _ = &mut shutdown_rx => {
                         let mut senders = Vec::new();
 
+                        // Prevent new messages from being sent.
+                        rx.close();
```

Comment:
Let's say right after this closes a new webhook request comes in and notices the channel has closed. So there are two threads running: the remaineder of the function here and a new webhook channel. Is there a correctness or availability race between those?

Reply:
After the latest refactor, we should only enter this shutdown path if the source was dropped, or environmentd is shutting down, so there shouldn't be a correctness or availability issue here